### PR TITLE
Init with Hex String

### PR DESCRIPTION
**Pull request by bpolania**
Now BItn and BDouble can be initialized with hex strings:

```
BDouble.init("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", 16) will return 115792089237316195423570985008687907853269984665640564039457584007913129639935
```

Readme file was also updated

### CODE REVIEW

**Comment by twodayslate** [comment]
I would recommend adding test cases and not changing the tab layout. Additionally, it would make more sense with base: in the initializer?

**Comment by twodayslate** [comment]
`radix` was fine. I meant remove the `_` before it.

File: Tests/SMP Tests.swift [hunk 1]
```diff
@@ -168,7 +167,24 @@ public class SMP_Tests
...
+        //
+        
+        XCTAssert(BInt("ffff",16) == 65535)
+        XCTAssert(BInt("ff",16) == 255.0)
+        XCTAssert(BInt("ff",16) != 100.0)
+        XCTAssert(BInt("ffff",16)! > 255.0)
+        XCTAssert(BInt("f",16)! < 255.0)
+        XCTAssert(BInt("0",16)! <= 1.0)
+        XCTAssert(BInt("f")! >= 1.0)
+
```
Comment (twodayslate) [inline Tests/SMP Tests.swift]: what happens when radix isn't 16? what happens if the string isn't hex?

Comment (bpolania) [reply Tests/SMP Tests.swift]: Got it. Will add cases gradually, I can only put only so much hours on this =)

**Comment by bpolania** [comment]
I added test cases for non radix 16 number systems, but it came to my attention that a non-base-16 system can't be considered "hexadecimal" or "hex", so perhaps the name of the branch may seem a little misleading. For that reason I'll update comments to the functions referencing "mathematical numerical systems of base" instead of just "hex".

File: README.md [hunk 2]
```diff
@@ -36,18 +36,21 @@ This subsection explains the usage of BInt and BDouble
 BInt(Int)
 BInt(UInt)
 BInt(String)?
+BInt(String,radix: Int)?
```
Comment (twodayslate) [inline README.md]: space between command and radix

File: Tests/SMP Tests.swift [hunk 3]
```diff
@@ -168,7 +167,32 @@ public class SMP_Tests
...
+        XCTAssert(BInt("f")! >= 1.0)
+        XCTAssert(BInt("rfff",radix:16) == nil)
+        
+        XCTAssert(BInt("44",radix:5) == 68.0)
+        XCTAssert(BInt("44",radix:5) != 100.0)
+        XCTAssert(BInt("321",radix:5)! > 255.0)
+        XCTAssert(BInt("3",radix:5)! < 255.0)
+        XCTAssert(BInt("0",radix:5)! <= 1.0)
+        XCTAssert(BInt("4",radix:5)! >= 1.0)
+        XCTAssert(BInt("rfff",radix:5) == nil)
```
Comment (twodayslate) [inline Tests/SMP Tests.swift]: add tests for popular radixes: 2, 8, 10, 16
add tests with/without prefixes (0b, 0o, 0x)

**Comment by mkrd** [comment]
Looks good so far, but I don't want to merge yet, because the diff in SMP.swift is totally messed up. Maybe your IDE converted all tabs to spaces?

**Comment by bpolania** [comment]
Yeah I saw that, but I'm using XCode so I'm not sure why this is happening, will try to fix during the weekend.

**Comment by bpolania** [comment]
OK, I think the Diff issue in the test files ise fixed now.

**Comment by mkrd** [comment]
Hm, I still see 2600+ lines changed in the file "SMP.swift". Sorry, but I can't merge that, it will mess up the diff

**Comment by bpolania** [comment]
Yes, this is a tricky one because is not causing any conflicts so they're hard to see, I think I fixed most of them, can you please check on your side?

**Comment by twodayslate** [comment]
I would recommend using the project included with this project instead of using your own or manually adding SMP.swift to another project.

**Comment by bpolania** [comment]
Well, nevermind dude, I think people will benefit for being able to use hex strings, but is up to you. Good luck!

**Comment by mkrd** [comment]
Yes, this feature is definitely benefical, you're absolutely right. Thanks to @twodayslate for fixing the diff problem 👍

**Comment by mkrd** [comment]
@bpolania Maybe your editor is configured to automatically convert tabs to spaces? You can turn that off in the settings

**Pull request closed without merging**